### ISSUE TITLE
Loginflow SSO for Nextcloud

### ISSUE THREAD

**Issue by sunjam** [super-productivity/super-productivity#914]
![image](https://user-images.githubusercontent.com/1787238/108604236-8082a680-7361-11eb-8f9c-89e89ee199b9.png)

I know you are super busy and already handling plenty.  I'd like to humbly request [LoginFlow support](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2) for Nextcloud on Desktop as a way to ease setting up WebDAV #599 and CalDAV #626.  Of course you already support these as a platform agnostic option.

- [Here is documentation](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2), which can apply to Desktop, Mobile Clients and Web Apps.
  - I understand there is an [outstanding issue with CORS on most platforms](https://github.com/nextcloud/server/issues/3131).

Perhaps one day it will be possible to support this and smooth out integration to be as simple as clicking "Grant Access" for any user already logged into Nextcloud on their browser, desktop, or mobile app.  Greatly simplifies the process in the manner we expect from Google Drive and Dropbox (fyi [there is also Oauth support from Nextcloud](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/oauth2.html)).  Thank you for considering either way!!!

**Comment by johannesjo** [comment]
Thanks for opening this up @sunjam ! Would this also solve the cross origin problem?

**Comment by johannesjo** [comment]
Still relevant, I think.

**Comment by sunjam** [comment]
It is different than CORS.  afaik You are able to authenticate against the
Nextcloud app on a mobile device, which is a very simple "allow this app"
process.

**Comment by Steccas** [comment]
This is important, and I'm willing to help

**Comment by johannesjo** [comment]
Glad to hear it @Steccas !  Most of the relevant existing code can be found here:
https://github.com/johannesjo/super-productivity/tree/master/src/app/imex/sync/web-dav

What else do you need to get started?

**Comment by EngelPika32** [comment]
Still relevant^ (though, it's probably low priority).

**Comment by sunjam** [comment]
Open please

**Comment by rokejulianlockhart** [comment]
Please reopen this issue. I literally can't use this application, because I use a non-self-hosted instance (https://wim.nl.tab.digital) where I don't possess permission to install the necessary Nextcloud https://apps.nextcloud.com/apps/webapppassword app.

**Comment by sunjam** [comment]
open please

**Comment by RokeJulianLockhart** [comment]
> https://github.com/johannesjo/super-productivity/issues/914#issuecomment-2062845266

@GitHub-Actions, it remains relevant.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/johannesjo/super-productivity/issues/914#issuecomment-2295523334

@GitHub-Actions, https://github.com/johannesjo/super-productivity/issues/914#issuecomment-2063926359

**Comment by RokeJulianLockhart** [comment]
> https://github.com/johannesjo/super-productivity/issues/914#issuecomment-2550128635

@GitHub-Actions, it remains relevant.

**Comment by beedell-roke** [comment]
Yes, it's relevant. Stale Bot, I shall consume your lineage if ever I locate the server that hosts you.

**Comment by qualitaetsdefensive** [comment]
I just started using Super Productivity last week and I love it, it helps me so much to structure myself. There is however one big caveat for me: I cannot connect to the deck on the nextcloud of my work since we log into nextcloud using OAuth. Or am I missing something? Would be greatly appreciated if another login process than Username / Password for deck is possible!
